src/main/java/sg/ncl/MainController.java

```diff
@@ -254,6 +257,18 @@ public String testbedInformation(Model model) throws IOException {
         return "testbedInformation";
     }
 
+    // get all the nodes' status
+    // there are three types of status
+    // "free" : node is free
+    // "in_use" : node is in use
+    // "reload" : node is in process of freeing
+    @RequestMapping("/testbedNodesStatus")
+    public String testbedNodesStatus(Model model) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)